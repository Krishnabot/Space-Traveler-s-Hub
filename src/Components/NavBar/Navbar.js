import { NavLink } from 'react-router-dom';
import logo from '.././images/logo.png';
import './Navbar.modules.css';

const NavBar = () => (
  <div className="nav">
    <div className="nav-bar">
      <img className="logo" src={logo} alt="planet" />
      <h1>Space Traveler&apos;s Hub</h1>
    </div>
    <NavLink to="/">Rockets</NavLink>
    <NavLink to="/Missions">Missions</NavLink>
    <NavLink to="/MyProfile">My Profile</NavLink>
  </div>
);

export default NavBar;
