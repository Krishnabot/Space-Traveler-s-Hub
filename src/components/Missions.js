import React from 'react';
import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import styles from './Missions.module.css';
import { joinMissions } from '../redux/Missions/Missions';

const Missions = (props) => {
  const { mission } = props;
  const dispatch = useDispatch();
  const clickHandeler = (event) => {
    dispatch(joinMissions({ id: event.target.id }));
  };
  return (
    <>
      <td className={styles.mission_name}>{mission.mission_name}</td>
      <td className={styles.mission_desc}>
        {mission.description}
        {mission.description}
        <span>
          <a href={mission.more}>See more.</a>
        </span>
      </td>
      <td className={styles.member_status}>
        {!mission.status && <p className={styles.inactive}>NOT A MEMBER</p>}
        {mission.status && <p className={styles.active}>Active Member</p>}
      </td>
      <td className={styles.mission_status}>
        {!mission.status && (
          <button
            type="button"
            className={styles.join}
            onClick={clickHandeler}
            id={mission.mission_id}
          >
            Join Mission
          </button>
        )}
        {mission.status && (
          <button
            type="button"
            className={styles.leave}
            onClick={clickHandeler}
            id={mission.mission_id}
          >
            Leave Mission
          </button>
        )}
      </td>
    </>
  );
};

Missions.propTypes = {
  mission: PropTypes.shape({
    mission_name: PropTypes.string,
    description: PropTypes.string,
    status: PropTypes.bool,
    mission_id: PropTypes.string,
    more: PropTypes.string,
  }).isRequired,
};

export default Missions;
