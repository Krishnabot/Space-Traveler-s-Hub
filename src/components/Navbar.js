/* eslint linebreak-style: ["error", "windows"] */
import React from 'react';
import { NavLink } from 'react-router-dom';
import styles from './Navbar.module.css';
import logo from '../assets/images/logo.png';

const Navbar = () => {
  const navLinks = [
    {
      navId: 1,
      path: '/Rockets',
      text: 'Rockets',
    },
    {
      navId: 2,
      path: '/Missions',
      text: 'Missions',
    },
    {
      navId: 3,
      path: 'MyProfile',
      text: 'My Profile',
    },
  ];

  return (
    <header className={styles.header}>
      <div>
        <img src={logo} className={styles.logo} alt="planet_logo" />
        <p className={styles.title}>Space Traveler&apos;s Hub</p>
      </div>
      <nav className={styles.nav}>
        <ul className={styles.navLink_wrap}>
          {navLinks.map((link) => (
            <li key={link.navId}>
              <NavLink className={styles.navLink} to={link.path}>
                {link.text}
              </NavLink>
            </li>
          ))}
        </ul>
      </nav>
    </header>
  );
};

export default Navbar;
