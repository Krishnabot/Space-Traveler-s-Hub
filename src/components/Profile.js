import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles/Profile.module.css';

const Profile = (props) => {
  const { name } = props;
  return (
    <div className={styles.profilewrap}>
      <p className={styles.name}>{name}</p>
    </div>
  );
};

export default Profile;

Profile.propTypes = {
  name: PropTypes.string,
}.isRequired;
