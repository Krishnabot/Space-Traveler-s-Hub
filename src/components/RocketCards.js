import React from 'react';
import styles from './RocketCards.module.css';

const RocketCards = () => (
  <div className={styles.cards}>
    <div className={styles.card1}>
      <img href="#" alt="Static" />
      Placeholder Image
      <h2>Falcon1</h2>
      <p>Some dummy text as placeholder</p>
      <button type="submit">Reserve Rocket</button>
    </div>
    <div className={styles.card2}>
      <img href="#" alt="Static" />
      Placeholder Image
      <h2>Falcon9</h2>
      <p>Some dummy text as placeholder</p>
      <button type="submit">Cancel Reservation</button>
    </div>
  </div>
);

export default RocketCards;
