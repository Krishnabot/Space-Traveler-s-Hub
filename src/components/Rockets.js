import React from 'react';
import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import styles from './RocketCards.module.css';
import { joinRockets } from '../redux/Rockets/Rockets';

const Rockets = (props) => {
  const { rocket } = props;
  const dispatch = useDispatch();
  const clickHandeler = (event) => {
    dispatch(joinRockets({ id: event.target.id }));
  };
  return (
    <>
      <div className={styles.flikr_images}><img  alt="Rocket Image"  src={src} /></div>
      <div className={styles.rocket_name}>{rocket.rocket_name}</div>
      <div className={styles.rocket_desc}>
        {rocket.description}
        {rocket.more}
      </div>
      <td className={styles.rocket_status}>
        {!rocket.status && (
        <button
          type="button"
          className="join"
          onClick={clickHandeler}
          id={rocket.id}
        >
          Reserve Rocket
        </button>
        )}
        {rocket.status && (
        <button
          type="button"
          className="leave"
          onClick={clickHandeler}
          id={rocket.id}
        >
          Cancel Reservation
        </button>
        )}
      </td>
    </>
  );
};

Rockets.propTypes = {
    src: PropTypes.string,
    rocket_name: PropTypes.string,
    description: PropTypes.string,
    status: PropTypes.bool,
    id: PropTypes.string,
    more: PropTypes.string,
};

Rockets.defaultProps = {
  rocket_name: 'default',
  description: 'Non',
  src: 'no-img',
  status: false,
  id: '0',
};

export default Rockets;
