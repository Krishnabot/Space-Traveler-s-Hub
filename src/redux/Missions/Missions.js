import { createSlice } from '@reduxjs/toolkit';
import fetchMissions from '../ConsumeAPI/FetchMission';

const missionsSlice = createSlice({
  name: 'missions',
  initialState: [],
  reducers: {
    joinMissions(state, action) {
      const newState = state.map((mission) => {
        if (mission.mission_id === action.payload.id) {
          return {
            ...mission,
            status: !mission.status,
          };
        }
        return {
          ...mission,
        };
      });
      return newState;
    },
  },
  extraReducers: (builder) => {
    builder.addCase(fetchMissions.fulfilled, (state, action) => action.payload.map((mission) => ({
      mission_id: mission.mission_id,
      mission_name: mission.mission_name,
      description: mission.description,
      more: mission.wikipedia,
      status: false,
    })));
  },
});

export const { joinMissions } = missionsSlice.actions;
export default missionsSlice.reducer;
