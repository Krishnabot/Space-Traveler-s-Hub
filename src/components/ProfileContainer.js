import React from "react";
import { Link } from "react-router-dom";
import { useSelector } from "react-redux";
import Profile from "./Profile";
import styles from "./ProfileContainer.module.css";

const ProfileContainer = () => {
  const missions = useSelector((state) => state.missions);
  const joinedMissions = missions.filter((mission) => mission.status === true);

  return (
    <div className={styles.container_wrap}>
      <div className={styles.misions}>
        <h2>My Missions</h2>
        {joinedMissions.length === 0 ? (
          <>
            <div className={styles.no_mission}>
              <span>No missions joined Yet! </span>
            </div>
            <Link to="/Missions">
              <button type="button" className={styles.join_mission}>
                Would You Like to join one?
              </button>
            </Link>
          </>
        ) : (
          joinedMissions.map((mission) => (
            <Profile key={mission.mission_id} name={mission.mission_name} />
          ))
        )}
      </div>
    </div>
  );
};

export default ProfileContainer;
