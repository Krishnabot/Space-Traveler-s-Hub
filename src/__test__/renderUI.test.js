import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Missions from '../components/Missions';
import MissionContainer from '../components/MissionsContainer';
import Rockets from '../components/Rockets';
import RocketCards from '../components/RocketCards';
import Navbar from '../components/Navbar';
import Profile from '../components/profile';
import ProfileContainer from '../components/ProfileContainer';
import store from '../redux/configureStore';

describe('Components rendering test', () => {
  it('whether it match snapshot for NavBar or not ', () => {
    const tree = render(
      <BrowserRouter>
        <Provider store={store}>
          <Navbar />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Whether it matches snapshot for the Mission Contaniner or not', () => {
    const tree = render(
      <BrowserRouter>
        <Provider store={store}>
          <MissionContainer />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Whether it matches snapshot for the Mission or not ', () => {
    const mission = {
      mission_name: 'Mission 1',
      description: 'This is the description for mission 1',
      more: 'A link',
      status: false,
    };

    const tree = render(
      <BrowserRouter>
        <Provider store={store}>
          <Missions mission={mission} />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Whether it matches snapshot for the RocketCards or not', () => {
    const tree = render(
      <BrowserRouter>
        <Provider store={store}>
          <RocketCards />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Whether it matches snapshot for the  Rockets or not', () => {
    const rocket = {
      rocket_name: 'Rocket  1',
      description: 'This is the description for mission 1',
      more: 'A link',
      status: false,
      image: 'A Link',
      id: 'Rocket 1',
    };

    const tree = render(
      <BrowserRouter>
        <Provider store={store}>
          <Rockets rocket={rocket} />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Whether it matches snapshot for the Profile Contaniner  or not', () => {
    const tree = render(
      <BrowserRouter>
        <Provider store={store}>
          <ProfileContainer />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Whether it matches snapshot for the Profile or not', () => {
    const tree = render(
      <BrowserRouter>
        <Provider store={store}>
          <Profile />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
