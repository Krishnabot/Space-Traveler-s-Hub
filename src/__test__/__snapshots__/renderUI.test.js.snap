// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components rendering test Whether it matches snapshot for the  Rockets or not 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flikr_images"
      >
        <img
          alt="Rocket  1"
          class="images"
          src="A Link"
        />
      </div>
      <div
        class="cardBody"
      >
        <h2
          class="rocket_name"
        >
          Rocket  1
        </h2>
        <div
          class="member_status"
        />
        <div
          class="rocket_description"
        >
          <span
            class="desc"
          >
            This is the description for mission 1
          </span>
        </div>
        <div
          class="rocket_status"
        >
          <button
            class="join"
            type="button"
          >
            <p
              class="bText"
            >
              Reserve Rocket
            </p>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flikr_images"
    >
      <img
        alt="Rocket  1"
        class="images"
        src="A Link"
      />
    </div>
    <div
      class="cardBody"
    >
      <h2
        class="rocket_name"
      >
        Rocket  1
      </h2>
      <div
        class="member_status"
      />
      <div
        class="rocket_description"
      >
        <span
          class="desc"
        >
          This is the description for mission 1
        </span>
      </div>
      <div
        class="rocket_status"
      >
        <button
          class="join"
          type="button"
        >
          <p
            class="bText"
          >
            Reserve Rocket
          </p>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Components rendering test Whether it matches snapshot for the Mission Contaniner or not 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="table_wrap"
      >
        <table
          class="table_head"
        >
          <thead>
            <tr>
              <th
                class="title"
              >
                Mission
              </th>
              <th
                class="description"
              >
                Description
              </th>
              <th
                class="status"
              >
                Status
              </th>
              <th
                class="missioncontroll"
              >
                Mission Controll
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="table_wrap"
    >
      <table
        class="table_head"
      >
        <thead>
          <tr>
            <th
              class="title"
            >
              Mission
            </th>
            <th
              class="description"
            >
              Description
            </th>
            <th
              class="status"
            >
              Status
            </th>
            <th
              class="missioncontroll"
            >
              Mission Controll
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Components rendering test Whether it matches snapshot for the Mission or not  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <td
        class="mission_name"
      >
        Mission 1
      </td>
      <td
        class="mission_desc"
      >
        This is the description for mission 1
        This is the description for mission 1
        <span>
          <a
            href="A link"
          >
            See more.
          </a>
        </span>
      </td>
      <td
        class="member_status"
      >
        <p
          class="inactive"
        >
          NOT A MEMBER
        </p>
      </td>
      <td
        class="mission_status"
      >
        <button
          class="join"
          type="button"
        >
          Join Mission
        </button>
      </td>
    </div>
  </body>,
  "container": <div>
    <td
      class="mission_name"
    >
      Mission 1
    </td>
    <td
      class="mission_desc"
    >
      This is the description for mission 1
      This is the description for mission 1
      <span>
        <a
          href="A link"
        >
          See more.
        </a>
      </span>
    </td>
    <td
      class="member_status"
    >
      <p
        class="inactive"
      >
        NOT A MEMBER
      </p>
    </td>
    <td
      class="mission_status"
    >
      <button
        class="join"
        type="button"
      >
        Join Mission
      </button>
    </td>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Components rendering test Whether it matches snapshot for the Profile Contaniner  or not 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container_wrap"
      >
        <div
          class="missions"
        >
          <h2>
            My Missions
          </h2>
          <div
            class="no_mission"
          >
            <span>
              No missions joined Yet! 
            </span>
          </div>
          <a
            href="/Missions"
          >
            <button
              class="join_mission"
              type="button"
            >
              Would You Like to join one?
            </button>
          </a>
        </div>
        <div
          class="rockets"
        >
          <h2>
            Rockets
          </h2>
          <div
            class="no_rocket"
          >
            <span>
              No Rocket Reserved Yet! 
            </span>
          </div>
          <a
            href="/Rockets"
          >
            <button
              class="join_rocket"
              type="button"
            >
              Would You Like to Reserve One?
            </button>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container_wrap"
    >
      <div
        class="missions"
      >
        <h2>
          My Missions
        </h2>
        <div
          class="no_mission"
        >
          <span>
            No missions joined Yet! 
          </span>
        </div>
        <a
          href="/Missions"
        >
          <button
            class="join_mission"
            type="button"
          >
            Would You Like to join one?
          </button>
        </a>
      </div>
      <div
        class="rockets"
      >
        <h2>
          Rockets
        </h2>
        <div
          class="no_rocket"
        >
          <span>
            No Rocket Reserved Yet! 
          </span>
        </div>
        <a
          href="/Rockets"
        >
          <button
            class="join_rocket"
            type="button"
          >
            Would You Like to Reserve One?
          </button>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Components rendering test Whether it matches snapshot for the Profile or not 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="profilewrap"
      >
        <p
          class="name"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="profilewrap"
    >
      <p
        class="name"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Components rendering test Whether it matches snapshot for the RocketCards or not 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="cards"
      >
        <div />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="cards"
    >
      <div />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Components rendering test whether it match snapshot for NavBar or not  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
      >
        <div>
          <img
            alt="planet_logo"
            class="logo"
            src="logo.png"
          />
          <p
            class="title"
          >
            Space Traveler's Hub
          </p>
        </div>
        <nav
          class="nav"
        >
          <ul
            class="navLink_wrap"
          >
            <li>
              <a
                class="navLink"
                href="/Rockets"
              >
                Rockets
              </a>
            </li>
            <li>
              <a
                class="navLink"
                href="/Missions"
              >
                Missions
              </a>
            </li>
            <li>
              <a
                class="navLink"
                href="/MyProfile"
              >
                MyProfile
              </a>
            </li>
          </ul>
        </nav>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
    >
      <div>
        <img
          alt="planet_logo"
          class="logo"
          src="logo.png"
        />
        <p
          class="title"
        >
          Space Traveler's Hub
        </p>
      </div>
      <nav
        class="nav"
      >
        <ul
          class="navLink_wrap"
        >
          <li>
            <a
              class="navLink"
              href="/Rockets"
            >
              Rockets
            </a>
          </li>
          <li>
            <a
              class="navLink"
              href="/Missions"
            >
              Missions
            </a>
          </li>
          <li>
            <a
              class="navLink"
              href="/MyProfile"
            >
              MyProfile
            </a>
          </li>
        </ul>
      </nav>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
